eox_core/api/v1/permissions.py

```diff
@@ -32,10 +32,22 @@ def load_permissions():
 class EoxCoreAPIPermission(permissions.BasePermission):
     """
     Defines a custom permissions to access eox-core API
+    This permissions makes sure that a token is created with the client credentias of the same site is being used on
     """
 
     def has_permission(self, request, view):
         """
         To grant accees, checks if the request user either can call eox-core API or if it's an admin user.
         """
+        try:
+            allowed_for_site = request.user.is_staff or request.get_host() in request.auth.client.url
+        except Exception:  # pylint: disable=broad-except
+            allowed_for_site = False
+
+        if not allowed_for_site:
+            # If we get here either someone is using a token created on one site in a different site
+            # Or there was a missconfiguration of the oauth client.
+            # To prevent leaking important information we return the most basic message
+            raise exceptions.NotAuthenticated(detail="Invalid token")
```

Comment:
Maybe, you can raise this exception on line 45, since 'allowed_for_site' is only used to raise this exception. @felipemontoya

Reply:
I had that first, but when I added the user.is_staff check the line's logic became hard to follow